src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.Icu.cs

```diff
@@ -291,18 +291,29 @@ private unsafe string IcuGetTimeFormatString(bool shortFormat)
             Debug.Assert(!GlobalizationMode.UseNls);
             Debug.Assert(_sWindowsName != null, "[CultureData.GetTimeFormatString(bool shortFormat)] Expected _sWindowsName to be populated already");
 
-            char* buffer = stackalloc char[ICU_ULOC_KEYWORD_AND_VALUES_CAPACITY];
-
-            bool result = Interop.Globalization.GetLocaleTimeFormat(_sWindowsName, shortFormat, buffer, ICU_ULOC_KEYWORD_AND_VALUES_CAPACITY);
-            if (!result)
+            ReadOnlySpan<char> span;
+#if TARGET_MACCATALYST || TARGET_IOS || TARGET_TVOS
+            if (GlobalizationMode.Hybrid)
             {
-                // Failed, just use empty string
-                Debug.Fail("[CultureData.GetTimeFormatString(bool shortFormat)] Failed");
-                return string.Empty;
+                string res = Interop.Globalization.GetLocaleTimeFormatNative(_sWindowsName, shortFormat);
```

Comment:
Just to confirm, `Interop.Globalization.GetLocaleTimeFormatNative` returns format patterns like ICU patterns, right?

Reply:
Yes, we get the data using https://developer.apple.com/documentation/foundation/nsdateformatterstyle/nsdateformattershortstyle and https://developer.apple.com/documentation/foundation/nsdateformatterstyle/nsdateformattermediumstyle which is equivalent of ICU patterns.